meerk40t/gui/panes/navigationpanels.py

```diff
@@ -933,104 +1029,148 @@ def matrix_updated(self):
+    def on_translate_down_1(self, event=None):  # wxGlade: Navigation.<event_handler>
         dx = 0
         dy = self.context.navigate_jog
-        self.context("translate %f %f\n" % (dx, dy))
-        self.matrix_updated()
+        self.translateit(dx, dy)
+
+    def on_translate_down_10(self, event=None):  # wxGlade: Navigation.<event_handler>
+        dx = 0
+        dy = self.context.navigate_jog * 10
+        self.translateit(dx, dy)
 
     def on_reset(self, event=None):  # wxGlade: Navigation.<event_handler>
         self.context("reset\n")
         self.matrix_updated()
 
-    def on_rotate_ccw(self, event=None):  # wxGlade: Navigation.<event_handler>
-        spooler, input_driver, output = self.context.registered[
-            "device/%s" % self.context.root.active
-        ]
-        x = input_driver.current_x if input_driver is not None else 0
-        y = input_driver.current_y if input_driver is not None else 0
-        self.context("rotate %fdeg %f %f\n" % (-5, x, y))
-        self.matrix_updated()
+    def on_rotate_ccw_5(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = -5.0
+        self.rotateit(angle)
 
-    def on_rotate_cw(self, event=None):  # wxGlade: Navigation.<event_handler>
-        spooler, input_driver, output = self.context.registered[
-            "device/%s" % self.context.root.active
-        ]
-        x = input_driver.current_x if input_driver is not None else 0
-        y = input_driver.current_y if input_driver is not None else 0
-        self.context("rotate %fdeg %f %f\n" % (5, x, y))
-        self.matrix_updated()
+    def on_rotate_cw_5(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = 5.0
+        self.rotateit(angle)
+
+    def on_rotate_ccw_90(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = -90.0
+        self.rotateit(angle)
+
+    def on_rotate_cw_90(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = 90.0
+        self.rotateit(angle)
+
+    def skewed_value(self, stxt):
+        tau = pi * 2
+        try:
+            stxt = stxt.lower()
+            if stxt.endswith("deg"):
+                angle = tau * float(stxt[:-3]) / 360.0
+                valu = tan(angle)
+            elif stxt.endswith("grad"):
+                angle = tau * float(stxt[:-4]) / 400.0
+                valu = tan(angle)
+            elif stxt.endswith(
+                "rad"
+            ):  # Must be after 'grad' since 'grad' ends with 'rad' too.
+                angle = float(stxt[:-3])
+                valu = tan(angle)
+            elif stxt.endswith("turn"):
+                angle = tau * float(stxt[:-4])
+                valu = tan(angle)
+            elif stxt.endswith("%"):
+                angle = tau * (float(stxt[:-1]) / 100.0)
+                valu = tan(angle)
+            else:
+                valu = float(stxt)
+        except:
```

Comment:
An `except` statement without one or more exceptions explicitly listed is really not a good idea.  

I would remove the try/except from here and wrap it around the statements that call `skewed_value` below - and if there is an exception, NOT do the transformation but instead indicate an error someone by e.g. red colour.

Reply:
On the text-element thingy... I would prefer to detach the matrix-textfield part from the graphical button part and expose it as a panel of it's own. This would bring more flexibility IMHO. Happy to do it (in a separate PR).